### ISSUE TITLE
no-unused-prop-types false positive on prop accessed within map call

### ISSUE THREAD

**Issue by alangpierce** [yannickcr/eslint-plugin-react#1347]
Strangely, I'm getting a no-unused-prop-types error for `b` but not `a` or `c` in this code:
```js
const MyComponent = (props) => (
  <div>
    {props.a.map(() => (
      <div>
        {props.b}
        {props.c}
      </div>
    ))}
  </div>
);
MyComponent.propTypes = {
  a: React.PropTypes.arrayOf(React.PropTypes.string),
  b: React.PropTypes.string,  // Error: 'b' PropType is defined but prop is never used (react/no-unused-prop-types)
  c: React.PropTypes.string,
};
```

**Comment by ljharb** [comment]
I believe this is fixed in master but not yet released.

A (better anyways) workaround is to destructure all props at the top of the function.

**Comment by leethree** [comment]
@ljharb when is the next release scheduled? thanks!

**Comment by ljharb** [comment]
That's up to @yannickcr.
